Make server work under non-admin user on Windows

### Server Version

```
0.1.0
```
(launched as regular, non-admin user on Windows 10 Pro)

### Terraform Version

```
0.12.24
```

### Client Version

```
Sublime Text LSP v0.9.8
```

### Terraform Configuration Files

```hcl
provider "aws" {
}
```

### Log Output

```
2020/04/29 22:29:10 exec.go:172: Starting C:\ProgramData\chocolatey\bin\terraform.exe ["terraform" "version"] in "C:\\Users\\RADEKS~1\\AppData\\Local\\Temp"...
2020/04/29 22:29:10 exec.go:138: Waiting for command to finish ...
2020/04/29 22:29:10 rpc_logger.go:45: Error for "initialize" (ID 1): [-32098] failed to get version: terraform (pid 5008) exited (code 1): exit status 1
stdout: ""
stderr: "Couldn't setup logging tempfile: open C:\\Windows\\terraform-log151023907: Access is denied."
```

### Expected Behavior

`terraform version` should've run successfully, with log file being created in temporary directory.

### Actual Behavior

Server stopped because Terraform stopped because Terraform couldn't create the temp log file in `C:\Windows` (which is the [default `TempDir` location](https://golang.org/pkg/os/#TempDir) if no ENV variables are available, as Windows doesn't seem to have a static system-wide temp dir? 😱 ).

### Steps to Reproduce

1. Open a folder with the above config
2. Open file with the above config from that folder
3. Watch log

--- 

### Proposal

We just need to whitelist a few more ENV variables here

https://github.com/hashicorp/terraform-ls/blob/4bb04ee41d984629e4e7f456b8f7f9405d8e253e/internal/terraform/exec/exec.go#L98-L109

--- a/internal/terraform/exec/exec.go
+++ b/internal/terraform/exec/exec.go
@@ -19,6 +19,12 @@ import (
 	"github.com/hashicorp/terraform-ls/logging"
 )
 
+// Environment variables to pass through to Terraform
+var passthroughEnvVars = []string{
+	// This allows Terraform to find custom-built providers
+	"HOME", "USER",
+}
+
 // cmdCtxFunc allows mocking of Terraform in tests while retaining
 // ability to pass context for timeout/cancellation
 type cmdCtxFunc func(context.Context, string, ...string) *exec.Cmd
@@ -100,12 +106,10 @@ func (e *Executor) cmd(args ...string) (*command, error) {
 	// so we don't need to ask checkpoint for upgrades.
 	cmd.Env = append(cmd.Env, "CHECKPOINT_DISABLE=1")
 
-	// This allows Terraform to find custom-built providers
-	if v := os.Getenv("HOME"); v != "" {
-		cmd.Env = append(cmd.Env, "HOME="+v)
-	}
-	if v := os.Getenv("USER"); v != "" {
-		cmd.Env = append(cmd.Env, "USER="+v)
+	for _, key := range passthroughEnvVars {
+		if value := os.Getenv(key); value != "" {
+			cmd.Env = append(cmd.Env, key+"="+value)
+		}
 	}
 
 	if e.execLogPath != "" {

--- a/internal/terraform/exec/exec.go
+++ b/internal/terraform/exec/exec.go
@@ -22,7 +22,11 @@ import (
 // Environment variables to pass through to Terraform
 var passthroughEnvVars = []string{
 	// This allows Terraform to find custom-built providers
-	"HOME", "USER",
+	"HOME", "USER", "USERPROFILE",
+	// This allows Terraform to create crash log in the desired temp directory
+	// os.TempDir would otherwise fall back to C:\Windows on Windows
+	// which has no write permissions for non-admins
+	"TMPDIR", "TMP", "TEMP",
 }
 
 // cmdCtxFunc allows mocking of Terraform in tests while retaining

